### ISSUE TITLE
[Bug]: Search did not return enough result after segment compaction

### ISSUE THREAD

**Issue by MrPresent-Han** [milvus-io/milvus#33227]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:master
- Deployment mode(standalone or cluster):
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

topK=10, but only get 9 entities after searching for e2e test:
testcases/test_compaction.py::TestCompactionOperation::test_compact_during_insert

### Expected Behavior

get correct number

### Steps To Reproduce

```markdown
testcases/test_compaction.py::TestCompactionOperation::test_compact_during_insert
```

**Comment by xiaofan-luan** [comment]
this could happen with duplicated pk

**Comment by yanliang567** [comment]
@MrPresent-Han is this the same to  #33297
/assign @MrPresent-Han 
/unassign

**Comment by congqixia** [comment]
Shall be fixed by #33359
